### PR TITLE
Make device mod public

### PR DESCRIPTION
**Pull request by jakalkhalili**
As mentioned in issue #136, I'd like to make `device` mod to be accessible when **boringtun** is used as library.

### CODE REVIEW

**Comment by vkrasnov** [comment]
Looks like you need to change a few use statements too.
Also can you gate it to not(windows)

**Comment by vkrasnov** [comment]
Sorry, looks like you fixed while I was answering, still need to gate windows out

**Comment by jakalkhalili** [comment]
I think the integration tests for macOS should be excluded too

File: src/device/mod.rs [hunk 1]
```diff
@@ -38,17 +38,17 @@ use std::sync::Arc;
...
+use crate::device::dev_lock::*;
 use crate::noise::errors::*;
 use crate::noise::handshake::parse_handshake_anon;
 use crate::noise::rate_limiter::RateLimiter;
 use crate::noise::*;
-use crate::peer::*;
-use crate::poll::*;
-use crate::tun::*;
-use crate::udp::*;
+use crate::device::peer::*;
```
Comment (vkrasnov) [inline src/device/mod.rs:48]: I think that instead of appending crate you can remove device. i.e. `use peer::*`, `use noise::*`.

**Comment by vkrasnov** [comment]
>I think the integration tests for macOS should be excluded too

I think the problem is with docker on Azure, not with this PR

File: src/device/mod.rs (hunk 1 above)
Comment (jakalkhalili) [reply src/device/mod.rs]: Okay, I'll refactor it

**Comment by jakalkhalili** [comment]
@vkrasnov Tests passed, but not for macOS. What should I do? I think about excluding macOS from integration tests because the Docker will not be added to the macOS agent. That's a good idea? What do you think?

**Comment by vkrasnov** [comment]
I tested locally and tests pass, I will need to check what the problem is on Azure. For now I will exempt the macOS check.

**Pull request merged**